### ISSUE TITLE
Can't stream rtsp feeds from across the internet

### ISSUE THREAD

**Issue by leggettc18** [k-yle/rtsp-relay#15]
Good Evening,

I was trying to test this out, as we were looking into using it to replace an existing setup that no longer works thanks to an upgraded NVR (previous one did not use RTSP, and had an api endpoint to get images instead. New one uses RTSP). It worked great for local streams, but when I port forwarded the RTSP streams from the customer to test with the actual cameras. It would not work. I have verified that I can access them when port forwarded via VLC, and I can access those cameras locally (set up a VPN connection to test).

Now in this instance we would be running this on a server on the same network as those cameras anyway, so its not a deal breaker. I would like to know why it wasn't working though.

**Comment by k-yle** [comment]
hi @leggettc18, some server infrastructure can block the WebSocket (`ws://`) data for the video feed which is sent over the same port as the `http://` or `https://` data. 

In one of my setups I use the [caddy webserver](https://caddyserver.com/) which used to [require you to manually allow websocket data](https://caddy.community/t/caddy-v2-how-to-proxy-websoket-v2ray-websocket-tls/7040).

I believe one of the things the port-forwarder needs to support is the `Upgrade:` and `Connection:` HTTP headers. 

You could test/debug this by opening the network tab of the dev tools and comparing if the [HTTP 101 requests](https://httpstatuses.com/101) are all successful and whether the headers are the same via your proxy vs locally

**Comment by leggettc18** [comment]
Yeah I got it figured out I think at this point. In my dev environment, what was actually getting blocked was the `rtsp` streams, I replaced them with `rstps` versions and then it worked fine. Part of what confused me admittedly is that I thought `rtsps` wasn't working either, but it turned out I hadn't port forwarded the port for those (whoops). I did have to swap `ws://` for `wss://` after deploying as well. So in other words you were spot on. Thanks for the help though.
